File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let tree = [];
    if(root == null){
        return tree;
    }
    let deque = [];
    deque.push(root);

    while(deque.length > 0){
        let h = [];
        let a = deque.length;
        for(let i = 0; i < a; i++){
            let temp = deque.shift();
            h.push(temp.val);
            if(temp.left != null){
                deque.push(temp.left);
            }
            if(temp.right != null){
                deque.push(temp.right);
            }
        }
        tree.push(h);
    }
    return tree;
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        tree = []
        if not root:
            return tree
        queue = deque()
        queue.append(root)
        while(len(queue) > 0):
            hold = []
            for i in range(len(queue)):
                h = queue.popleft()
                hold.append(h.val)
                if h.left:
                    queue.append(h.left)
                if h.right:
                    queue.append(h.right)
            tree.append(hold)
        return tree
